### PR TITLE
showCollisionBoxes fix

### LINKED ISSUE #4647: Collision boxes don't appear until you move map
**mapbox-gl-js version**: All versions above 0.33.0

[Minimal demo of bug](https://jsbin.com/quqofec/edit?html,js,console,output)

No errors or warnings.

### Steps to Trigger Behavior

 1. Apply `map.showCollisionBoxes = true` (happens on load in demo)
 2. No collision boxes appear
 3. Move map
 4. Collision boxes appear

### Expected Behavior

Collision boxes show up on `map.showCollisionBoxes = true`

### Actual Behavior

Collision boxes show up only after moving map when `map.showCollisionBoxes = true`

### PR DESCRIPTION
**Pull request by mollymerp**
fix #4647 

When I was refactoring the `data` events in #4347 it appeared that this `fire` was unnecessary because I didn't think to check the `showCollisionBoxes` feature. Now we have a test to catch something like that in the future 😅 

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Can you test the added integration test in native and add an ignore or skip if needed? 🙇

File: test/integration/render-tests/regressions/mapbox-gl-js#4647/style.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  "metadata": {
+    "test": {
+      "height": 64,
+      "width": 64,
+      "operations": [
+        [
+          "wait"
+        ],
+        [
+          "showCollisionBoxes"
```
Comment (jfirebaugh) [inline test/integration/render-tests/regressions/mapbox-gl-js#4647/style.json]: https://github.com/mapbox/mapbox-gl-js/blob/878008895104b67861ce73f65809f7f6f0ed7263/test/integration/render-tests/debug/collision/style.json#L8-L8 is similar, but with different terminology. Can we make consistent?

**Comment by mollymerp** [comment]
ty @jfirebaugh 👍  tested on native and made all collision box tests consistent.

**Comment by asheemmamoowala** [comment]
🍏

**Comment by mollymerp** [comment]
@asheemmamoowala thanks! I still need an Official ✅ Review to be able to merge this 😸

**Pull request merged**